tensorflow_serving/model_servers/server.cc

```diff
@@ -285,7 +287,9 @@ Status Server::BuildAndStart(const Options& server_options) {
   options.flush_filesystem_caches = server_options.flush_filesystem_caches;
   options.allow_version_labels_for_unavailable_models =
       server_options.allow_version_labels_for_unavailable_models;
-
+  options.total_model_memory_limit_bytes = std::min(
```

Comment:
update `total_model_memory_limit_bytes` only when `server_options.total_model_memory_limit_megabytes` is not zero (implying someone changed the value via command line). this ensures that default limit continues to be applied as before. your current change assumes that the default is unit64_max.